base/namedtuple.jl

```diff
@@ -89,6 +89,7 @@ firstindex(t::NamedTuple) = 1
 lastindex(t::NamedTuple) = nfields(t)
 getindex(t::NamedTuple, i::Int) = getfield(t, i)
 getindex(t::NamedTuple, i::Symbol) = getfield(t, i)
+getindex(t::NamedTuple, r::AbstractVector) = ((getfield(t, ri) for ri in r)...,)
```

Comment:
Maybe `map(i -> t[i], r)`?

Reply:
`typeof` of the values are the same, in the current implementation too.

Reply:
More over, see this.
![image](https://user-images.githubusercontent.com/33925317/54832178-ad476280-4ce1-11e9-870b-4df6f276d5a9.png)
`map` is taking more allocations and more time, than the current implementation!

Reply:
You are measuring compilation time and benchmarking in global scope. I would suggest reading through the performance tips in the manual and the manual for BenchmarkTools.jl to get good info about benchmarking Julia code.

Reply:
> `typeof` of the values are the same, in the current implementation too.

Yes - I was more referring to `typeof` the output container rather than `eltype` of the output container, if that makes sense?

Reply:
Ok, got it!

Reply:
@KristofferC thanks for guiding me, I didn't have any idea regarding this aspect (noobie here), but I guess now I understand it!